package.json

```diff
@@ -28,7 +28,25 @@
   "devDependencies": {
     "@types/chai": "^4.2.14",
     "@types/mocha": "^8.2.0",
+    "browser-sync": "^2.26.7",
+    "browser-sync-webpack-plugin": "^2.0.1",
     "chai": "^4.2.0",
+    "codemirror": "^5.59.4",
+    "gulp": "^4.0.2",
+    "gulp-autoprefixer": "^7.0.1",
+    "gulp-bower": "0.0.15",
+    "gulp-eslint": "^6.0.0",
+    "gulp-istanbul": "^1.1.3",
+    "gulp-less": "^4.0.1",
+    "gulp-livereload": "^4.0.2",
+    "gulp-minify": "^3.1.0",
+    "gulp-notify": "^3.2.0",
+    "gulp-plumber": "^1.2.1",
+    "gulp-ruby-sass": "^4.0.0",
+    "gulp-sass": "^4.1.0",
```

Comment:
Are all of these really necessary? For example, I just saw [gulp-ruby-sass](https://www.npmjs.com/package/gulp-ruby-sass) is deprecated (2 years ago) in favor of only using gulp-sass. I've never used gulp before, so this is more a question to clarify. In particular, I also wonder how gulp as a stream builder interacts here with webpack as a bundler (e.g. gulp is only being used for theming and is better choice than webpack/works with webpack because X).

Reply:
I added the packages from another project of mine, at the time there were some features that are not completely migrated to `gulp-sass` so I needed to specify both. For the sake of this project I would not expect such issues. Will remove `gulp-ruby-sass`

Reply:
Can you say generally what the `gulp` packages are for? I'm not opposed to having several, but I don't recognize the majority of them.

Reply:
The most important usage for the gulp packages such as `gulp-sass` and `gulp-minify` are to assist us styling the page since we will be writing sass. Some packages such as `gulp-sourcemaps` can help us while debugging, especially when we are debugging styling with minified css. Others such as `gulp-eslint` are to help us with javascript, while I expect to use them in the future but they are currently not used by front-end

Reply:
Can we leave out the ones that we aren't using yet, just to keep the history of reasons we added things to `package.json` clear? Thanks for going through all this build stuff (typical web annoyances).

Reply:
Since I'm a pretty new to webpack after Winston's help now the site should be using webpack instead of gulp, so all the gulp packages are now removed. Learned a lot from Winston's suggestions and branch, kudos to him!

Reply:
Awesome @shanliyu, glad I could help! note: might still need to cleanup the last couple files:
- theme/gulpfile.js
- theme/css/test.css
- theme/css/test.css.map
- theme/js/test.js